# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/tsa/slrn/link_subjects.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/art_sort.c	Mon Mar 11 15:22:35 2002
+++ ./src/art_sort.c	Thu Mar 14 19:02:45 2002
@@ -526,7 +526,7 @@
 	       {
 		  Slrn_Header_Type *rjh = relatives[j].h;
 		  
-		  if (Slrn_New_Subject_Breaks_Threads
+		  if ((Slrn_New_Subject_Breaks_Threads & 1)
 		      && (rih->subject != NULL)
 		      && (rjh->subject != NULL)
 		      && (0 != _art_subject_cmp (rih->subject, rjh->subject)))
@@ -877,7 +877,7 @@
 	       {
 		  Slrn_Header_Type *child, *last_child, *rparent;
 		  
-		  if (Slrn_New_Subject_Breaks_Threads
+		  if ((Slrn_New_Subject_Breaks_Threads & 1)
 		      && (h->subject != NULL)
 		      && (ref->subject != NULL)
 		      && (0 != _art_subject_cmp (h->subject, ref->subject)))
@@ -921,8 +921,11 @@
    
    /* Now perform sort on subject to catch those that have fallen through the
     * cracks, i.e., no references */
-   slrn_message_now (_("Linking articles with identical subjects ..."));
-   link_same_subjects ();
+   if (!(Slrn_New_Subject_Breaks_Threads & 2))
+     {
+	slrn_message_now (_("Linking articles with identical subjects ..."));
+	link_same_subjects ();
+     }
    
    /* Now link up others as sisters */
    h = Slrn_First_Header;
